test/extensions/filters/listener/common/fuzz/listener_filter_fakes.h

```diff
@@ -0,0 +1,95 @@
+#include "common/api/os_sys_calls_impl.h"
+#include "common/network/io_socket_handle_impl.h"
+#include "common/network/utility.h"
+
+#include "test/mocks/network/mocks.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+
+class FakeConnectionSocket : public Network::MockConnectionSocket {
+public:
+  FakeConnectionSocket()
+      : io_handle_(std::make_unique<Network::IoSocketHandleImpl>(42)), local_address_(nullptr),
+        remote_address_(nullptr) {}
+
+  ~FakeConnectionSocket() override { io_handle_->close(); }
+
+  Network::IoHandle& ioHandle() override { return *io_handle_; }
+
+  const Network::IoHandle& ioHandle() const override { return *io_handle_; }
+
+  void setLocalAddress(const Network::Address::InstanceConstSharedPtr& local_address) override {
+    local_address_ = local_address;
+    if (local_address_ != nullptr) {
+      addr_type_ = local_address_->type();
+    }
+  }
+
+  void setRemoteAddress(const Network::Address::InstanceConstSharedPtr& remote_address) override {
+    remote_address_ = remote_address;
+  }
+
+  const Network::Address::InstanceConstSharedPtr& localAddress() const override {
+    return local_address_;
+  }
+
+  const Network::Address::InstanceConstSharedPtr& remoteAddress() const override {
+    return remote_address_;
+  }
+
+  Network::Address::Type addressType() const override { return addr_type_; }
+
+  absl::optional<Network::Address::IpVersion> ipVersion() const override {
+    if (addr_type_ != Network::Address::Type::Ip) {
+      return absl::nullopt;
+    }
+
+    return local_address_->ip()->version();
+  }
+
+  void setRequestedApplicationProtocols(const std::vector<absl::string_view>& protocols) override {
+    application_protocols_.clear();
+    for (const auto& protocol : protocols) {
+      application_protocols_.emplace_back(protocol);
+    }
+  }
+
+  const std::vector<std::string>& requestedApplicationProtocols() const override {
+    return application_protocols_;
+  }
+
+  Api::SysCallIntResult getSocketOption(int level, int, void* optval, socklen_t*) const override {
+    switch (level) {
+    case SOL_IPV6:
+      static_cast<sockaddr_storage*>(optval)->ss_family = AF_INET6;
+      break;
+    case SOL_IP:
+      static_cast<sockaddr_storage*>(optval)->ss_family = AF_INET;
+      break;
+    default:
+      NOT_REACHED_GCOVR_EXCL_LINE;
+    }
+
+    return Api::SysCallIntResult{0, 0};
+  }
+
+private:
+  const Network::IoHandlePtr io_handle_;
+  Network::Address::InstanceConstSharedPtr local_address_;
+  Network::Address::InstanceConstSharedPtr remote_address_;
+  Network::Address::Type addr_type_;
```

Comment:
What is the default address type that this gets set to?

Reply:
I think it's set to IP by default - that's what was causing original_dst_fuzz_test to fail (segmentation fault) on invalid_scheme.